crates/chain/src/spk_client.rs

```diff
@@ -1,388 +1,552 @@
-            txids: Box::new(core::iter::empty()),
-            outpoints: Box::new(core::iter::empty()),
+            inner: Default::default(),
         }
     }
+}
 
-    /// Set the [`Script`]s that will be synced against.
-    ///
-    /// This consumes the [`SyncRequest`] and returns the updated one.
-    #[must_use]
-    pub fn set_spks(
-        mut self,
-        spks: impl IntoIterator<IntoIter = impl ExactSizeIterator<Item = ScriptBuf> + Send + 'static>,
+#[cfg(feature = "miniscript")]
+impl<K: Clone + Ord + core::fmt::Debug + Send + Sync> SyncRequestBuilder<(K, u32)> {
+    /// Add [`Script`]s that are revealed by the `indexer` of the given `spk_range` that will be
+    /// synced against.
+    pub fn revealed_spks_from_indexer(
+        self,
+        indexer: &crate::indexer::keychain_txout::KeychainTxOutIndex<K>,
+        spk_range: impl core::ops::RangeBounds<K>,
     ) -> Self {
-        self.spks = Box::new(spks.into_iter());
-        self
+        self.spks_with_labels(indexer.revealed_spks(spk_range))
     }
 
-    /// Set the [`Txid`]s that will be synced against.
-    ///
-    /// This consumes the [`SyncRequest`] and returns the updated one.
-    #[must_use]
-    pub fn set_txids(
-        mut self,
-        txids: impl IntoIterator<IntoIter = impl ExactSizeIterator<Item = Txid> + Send + 'static>,
+    /// Add [`Script`]s that are revealed by the `indexer` but currently unused.
+    pub fn unused_spks_from_indexer(
+        self,
+        indexer: &crate::indexer::keychain_txout::KeychainTxOutIndex<K>,
     ) -> Self {
-        self.txids = Box::new(txids.into_iter());
-        self
+        self.spks_with_labels(indexer.unused_spks())
+    }
+}
+
+impl SyncRequestBuilder<()> {
+    /// Add [`Script`]s that will be synced against.
+    pub fn spks(self, spks: impl IntoIterator<Item = ScriptBuf>) -> Self {
+        self.spks_with_labels(spks.into_iter().map(|spk| ((), spk)))
     }
+}
 
-    /// Set the [`OutPoint`]s that will be synced against.
+impl<SpkLabel> SyncRequestBuilder<SpkLabel> {
+    /// Set the initial chain tip for the sync request.
     ///
-    /// This consumes the [`SyncRequest`] and returns the updated one.
-    #[must_use]
-    pub fn set_outpoints(
-        mut self,
-        outpoints: impl IntoIterator<
-            IntoIter = impl ExactSizeIterator<Item = OutPoint> + Send + 'static,
-        >,
-    ) -> Self {
-        self.outpoints = Box::new(outpoints.into_iter());
+    /// This is used to update [`LocalChain`](crate::local_chain::LocalChain).
+    pub fn chain_tip(mut self, cp: CheckPoint) -> Self {
+        self.inner.chain_tip = Some(cp);
         self
     }
 
-    /// Chain on additional [`Script`]s that will be synced against.
-    ///
-    /// This consumes the [`SyncRequest`] and returns the updated one.
-    #[must_use]
-    pub fn chain_spks(
+    /// Add [`Script`]s coupled with an associated label that will be synced against.
+    pub fn spks_with_labels(
```

Comment:
I think an example could help for knowing how to include the `SpkLabel`

```rust
/// # Example
/// 
/// Sync revealed script pubkeys obtained from a [`KeychainTxOutIndex`].
///
/// ```rust
/// # use bdk_chain::spk_client::SyncRequest;
/// # use bdk_chain::indexer::keychain_txout::KeychainTxOutIndex;
/// # let indexer = KeychainTxOutIndex::<()>::default();
/// let indexed_spks = indexer.revealed_spks(..);
/// let request = SyncRequest::builder()
///     .spks_with_labels(indexed_spks)
///     .build();
/// ```
```